ucla_migration.drush.inc

```diff
@@ -3821,14 +3821,42 @@ EOQ;
      *   Valid path to a pdf file.
      */
     function _add_pdf_to_object($object, $pdf_path) {
-        $object->relationships->add(FEDORA_MODEL_URI, 'hasModel', 'islandora:sp_pdf');
-        $datastream = $object->constructDatastream('OBJ', 'M');
-        $datastream->label = "PDF Document";
-        $datastream->mimetype = "application/pdf";
-        $datastream->setContentFromFile($pdf_path);
-        $object->ingestDatastream($datastream);
+
+        // get pdf datastream from object if exist or else create a new one add this logic
+        watchdog('Ingest Success', 'In add/update pdf @file_name', array('@file_name' => $pdf_path), WATCHDOG_NOTICE);
+        $is_new_pdf_ds = FALSE;
+        if (isset($object['OBJ'])) {
+            $datastream = $object['OBJ'];
+        }
+        else {
+            $object->relationships->add(FEDORA_MODEL_URI, 'hasModel', 'islandora:sp_pdf');
```

Comment:
is this necessary? If the object already exists, it more than likely _already has_ the islandora:sp_pdf model

You can use the [relationships->get()](https://github.com/Islandora/islandora/wiki/Working-With-Fedora-Objects-Programmatically-Via-Tuque#example-of-retrieving-a-filtered-relationship) method to check if this is present or not

Reply:
Added a relationship check.